pandas/indexes/multi.py

```diff
@@ -667,15 +667,16 @@ def _has_complex_internals(self):
     @cache_readonly
     def is_monotonic(self):
 
-        # TODO
-        # this is unfortunate we end up tupelizing
-        # just to determine monotonicity :<
-
-        # fast-path
-        if not self.levels[0].is_monotonic:
-            return False
-
-        return Index(self.values).is_monotonic
+        def level_values(level):
+            unique = self.levels[level]
+            labels = self.labels[level]
+            return algos.take_1d(unique.values, labels,
+                                 fill_value=unique._na_value)
+
+        # reversed() because lexsort() wants the most significant key last.
+        values = [level_values(i) for i in reversed(range(len(self.levels)))]
+        sort_order = np.lexsort(values)
+        return Index(sort_order).is_monotonic
```

Comment:
@ssanderson this avoids the worst case scenario as this is a short-circuit operator (in cython)